pkg/netutils/interface.go

```diff
@@ -171,25 +171,27 @@ func isLoopbackOrPointToPoint(intf *net.Interface) bool {
 // getMatchingGlobalIP returns the first valid global unicast address of the given
 // 'family' from the list of 'addrs'.
 func getMatchingGlobalIP(addrs []net.Addr, family AddressFamily) (net.IP, error) {
-	if len(addrs) > 0 {
-		for i := range addrs {
-			glog.V(4).Infof("Checking addr  %s.", addrs[i].String())
-			ip, _, err := net.ParseCIDR(addrs[i].String())
-			if err != nil {
-				return nil, err
-			}
-			if memberOf(ip, family) {
-				if ip.IsGlobalUnicast() {
-					glog.V(4).Infof("IP found %v", ip)
-					return ip, nil
-				}
-				glog.V(4).Infof("Non-global unicast address found %v", ip)
-			} else {
-				glog.V(4).Infof("%v is not an IPv%d address", ip, int(family))
-			}
+	if len(addrs) == 0 {
+		return nil, nil
+	}
 
+	for i := range addrs {
+		glog.V(4).Infof("Checking addr  %s.", addrs[i].String())
+		ip, _, err := net.ParseCIDR(addrs[i].String())
+		if err != nil {
+			return nil, err
+		}
+		if memberOf(ip, family) {
```

Comment:
According to the logic, I consider that there is an alternative method which I usually used to write:

```
if ! memberOf(ip, family) {
     glog.V(4).Infof("%v is not an IPv%d address", ip, int(family))
     continue
}

if ip.IsGlobalUnicast() {
    glog.V(4).Infof("IP found %v", ip)
    return ip, nil
}
glog.V(4).Infof("Non-global unicast address found %v", ip)
```
I recommend this code style because it can deduce the nested level, but it seems like that the readability perhaps  has a decrease.

Reply:
LGTM. Could you also refactor `memberOf` to fast return pattern ?

Reply:
@fengzixu Yes, it is more idiomatic 👍